src/ws/mock-service.c

```diff
@@ -28,9 +28,11 @@
 #include <string.h>
 
 typedef struct {
+  TestFrobber *frobber;
```

Comment:
Can't you just directly submit the invocation without needing to keep this around?

Reply:
I'm afraid I don't know what you mean -- `test_frobber_complete_never_return()` needs both the object (TestFrobber) and the invocation data. The alternative would be to keep the individual `object` from `on_handle_never_return()` as well, and store a new struct in the GList (which contains object and invocation); but (1) that would make the whole bookeeping more complicated, and (2) it's always the same object anyway, so I found this a lot simpler.

Reply:
So I tested
```diff
--- src/ws/mock-service.c
+++ src/ws/mock-service.c
@@ -720,7 +720,7 @@ mock_data_free (gpointer data)
 
   /* answer all NeverReturn() calls now, to avoid service shutdown hang */
   for (GList *elem = mock_data->never_return_invocations; elem; elem = elem->next)
-    test_frobber_complete_never_return (mock_data->frobber, elem->data);
+    test_frobber_complete_never_return (NULL, elem->data);
   g_list_free_full (mock_data->never_return_invocations, g_object_unref);
 
   g_object_unref (mock_data->connection);
```

and this actually seems to work -- is that kosher? I.e. can I *rely* on these (auto-generated) handlers to not require `object`? If so, then I don't need to keep around `frobber`, and the patch would be simpler. But that feels a bit like pushing luck?

Reply:
Oh, I see the gdbus-codegen autogenerated code:
```
void
test_frobber_complete_never_return (
    TestFrobber *object G_GNUC_UNUSED,
    GDBusMethodInvocation *invocation)
```

and that seems to be the case for all generated test_frobber_complete_*() functions. So if you tell me that this is reliable and safe, I'll be happy to drop it!

Reply:
I mean that you can just call `g_dbus_method_invocation_return_value()`.

Reply:
Aah, I see -- that works fine, thanks! Updated.